src/ring_theory/ideal/operations.lean

```diff
@@ -1604,3 +1604,105 @@ begin
 end
 
 end ring_hom
+
+namespace double_quot
+variables {R : Type u} [comm_ring R] (I J : ideal R)
+
+lemma add_comm : I+J = J+I := _root_.add_comm I J
+
+-- a few lemmas to help shorten the proofs later
+lemma left_proj_quot_add_mk (x :R) (hx : x ∈ I) : ideal.quotient.mk (I+J) x = 0 :=
+ideal.quotient.eq_zero_iff_mem.2 (ideal.mem_sup_left hx)
+
+lemma right_proj_quot_add_mk (x : R) (hx : x ∈ J) :  ideal.quotient.mk (I+J) x = 0 :=
+ideal.quotient.eq_zero_iff_mem.2 (ideal.mem_sup_right hx)
+
+lemma in_ker_proj_to_add_left : I.map(ideal.quotient.mk(I+J)) = ⊥ :=
+begin
+  simp_rw [ideal.map, ideal.span_eq_bot, set.mem_image],
+  rintros y ⟨x, hx, rfl⟩,
+  exact left_proj_quot_add_mk I J x hx,
+end
+
+
+lemma in_ker_proj_to_add_right : J.map(ideal.quotient.mk(I+J)) = ⊥ :=
+by {rw add_comm, apply in_ker_proj_to_add_left}
+
+/-- define `quot_left_to_quot_sum` to be the obvious ring hom `R/I → R/(I+J)` -/
+def quot_left_to_quot_sum : I.quotient →+* (I+J).quotient :=
```

Comment:
A general comment; the lemma `ideal.add_eq_sup` basically tells lean "we want to use the spelling `I ⊔ J` instead of `I + J` everywhere". It would make sense to replace `+` with `⊔` in all your theorem and lemma statements.

Reply:
Ok ! I've made the changes.